mp/src/game/client/hud_damageindicator.cpp

```diff
@@ -0,0 +1,332 @@
+//========= Copyright 1996-2005, Valve Corporation, All rights reserved. ============//
+//
+// Purpose: Hud element that indicates the direction of damage taken by the player
+//
+//=============================================================================//
+
+#include "cbase.h"
+#include <vgui_controls/AnimationController.h>
+#include "hud_macros.h"
+#include "hudelement.h"
+#include "iclientmode.h"
+#include "view.h"
+#include "util/mom_util.h"
+
+using namespace vgui;
+
+// memdbgon must be the last include file in a .cpp file!!!
+#include "tier0/memdbgon.h"
+
+// Convars for customizing damage indicator
+static ConVar enabled("mom_hud_damageindicator", "1", FCVAR_CLIENTDLL | FCVAR_ARCHIVE | FCVAR_REPLICATED,
+                      "Enable or disable damage indicator.\n");
+static ConVar minwidth("mom_hud_damageindicator_minwidth", "20", FCVAR_CLIENTDLL | FCVAR_ARCHIVE | FCVAR_REPLICATED,
+                       "Minimum width of the damage indicator.\n");
+static ConVar maxwidth("mom_hud_damageindicator_maxwidth", "20", FCVAR_CLIENTDLL | FCVAR_ARCHIVE | FCVAR_REPLICATED,
+                       "Maximum width of the damage indicator \n");
+static ConVar minheight("mom_hud_damageindicator_minheight", "60", FCVAR_CLIENTDLL | FCVAR_ARCHIVE | FCVAR_REPLICATED,
+                        "Minimum height of the damage indicator.\n");
+static ConVar maxheight("mom_hud_damageindicator_maxheight", "120", FCVAR_CLIENTDLL | FCVAR_ARCHIVE | FCVAR_REPLICATED,
+                        "Maximum height of the damage indicator \n");
+static ConVar radius("mom_hud_damageindicator_radius", "120", FCVAR_CLIENTDLL | FCVAR_ARCHIVE | FCVAR_REPLICATED,
+                     "How far away the damage indicators are from the crosshair.\n");
+static ConVar lifetime("mom_hud_damageindicator_lifetime", "1", FCVAR_CLIENTDLL | FCVAR_ARCHIVE | FCVAR_REPLICATED,
+                       "How long the indicator stays on screen.\n");
+static ConVar dmgcolor("mom_hud_damageindicator_color", "980000", FCVAR_CLIENTDLL | FCVAR_ARCHIVE | FCVAR_REPLICATED,
+                       "Color of the damage indicator.\n");
+
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
+class CHudDamageIndicator : public CHudElement, public vgui::Panel
+{
+    DECLARE_CLASS_SIMPLE(CHudDamageIndicator, vgui::Panel);
+
+  public:
+    CHudDamageIndicator(const char *pElementName);
+    void Init(void);
+    void VidInit(void);
+    void Reset(void);
+    virtual bool ShouldDraw(void);
```

Comment:
Remove `(void)` from parenthesis, it's an idiom from the 90s we don't use anymore.